src/App.tsx

```diff
@@ -15,71 +20,124 @@ export const goods = [
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
+export class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  reset = () => {
+    goods.forEach(good => {
+      const element:HTMLElement | null = document.getElementById(`${good}`);
+
+      if (element) {
+        element.classList.remove('has-background-success-light');
+        const button:HTMLElement | null = element.querySelector('.button');
+
+        if (button && button.classList.contains('is-info')) {
+          button.classList.remove('is-info');
+          button.setAttribute('data-cy', 'AddButton');
+          button.textContent = '+';
+        }
+      }
+    });
+
+    this.setState({ selectedGood: null });
+  };
+
+  handleClick = (good: string) => {
+    const element:HTMLElement | null = document.getElementById(`${good}`);
+
+    if (element) {
+      if (element.classList.contains('has-background-success-light')) {
+        this.reset();
+      } else {
+        this.reset();
+
+        if (element) {
+          element.classList.add('has-background-success-light');
+          const button:HTMLElement | null = element.querySelector('.button');
+
+          if (button) {
+            goods.forEach(good => {
+              const overButton:HTMLElement | null = document.getElementById(`${good}`);
+
+              if (overButton) {
+                const otherButton:HTMLElement | null
+                = overButton.querySelector('.button');
+
+                if (otherButton) {
+                  otherButton.classList.remove('is-info');
+                  otherButton.setAttribute('data-cy', 'AddButton');
+                  otherButton.textContent = '+';
+                }
+              }
+            });
+
+            button.setAttribute('data-cy', 'RemoveButton');
+            button.classList.add('is-info');
+            button.textContent = '-';
+          }
+        }
+
+        this.setState({ selectedGood: good });
+      }
+    }
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <main className="section container">
+        {selectedGood ? (
+          <h1 className="title is-flex is-align-items-center">
+            {selectedGood}
+            {' '}
+            is selected
+
             <button
-              data-cy="AddButton"
+              data-cy="ClearButton"
               type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+              className="delete ml-3"
+              onClick={this.reset}
```

Comment:
now here we will change it to be:

`onClick={this.clear}` or `onClick={this.clearItem}` depending what name you will chose